experiments/reconstruct_shakespeare.py

```diff
@@ -35,6 +39,11 @@
     parser.add_argument("--nhead", type=int, default=8, help="Number of attention heads")
     parser.add_argument("--num_decoder_layers", type=int, default=6, help="Number of decoder layers")
     parser.add_argument("--dim_feedforward", type=int, default=2048, help="Dimension of the feedforward network")
+    parser.add_argument("--nembd", type=int, default=768, help="Dimension of the embedding layer for NanoGPT")
+    parser.add_argument("--nlayer", type=int, default=12, help="Number of layers for NanoGPT")
```

Comment:
Let's get rid of num_encoder_layers and num_decoder_layers and just substitute this in to make it consistent